src/main/scala/com/cognite/sdk/scala/common/OAuth2.scala

```diff
@@ -65,106 +153,52 @@ object OAuth2 {
   object ClientCredentialsProvider {
     def apply[F[_]](
         credentials: ClientCredentials,
-        refreshSecondsBeforeTTL: Long = 30
+        refreshSecondsBeforeTTL: Long = 30,
+        maybeCacheToken: Option[TokenState] = None
     )(
         implicit F: Async[F],
         clock: Clock[F],
         sttpBackend: SttpBackend[F, Any]
     ): F[ClientCredentialsProvider[F]] = {
-      val authenticate: F[TokenState] = {
-        val body = Map[String, String](
-          "grant_type" -> "client_credentials",
-          "client_id" -> credentials.clientId,
-          "client_secret" -> credentials.clientSecret,
-          // Aize auth flow requires this to be empty in general
-          "scope" -> credentials.scopes.mkString(" "), // Send empty scopes when it is not provided
-          // AAD auth flow requires this to be empty
-          "audience" -> credentials.audience.getOrElse(
-            ""
-          ) // Send empty audience when it is not provided
-        )
+      val authenticate: F[TokenState] =
         for {
-          response <- basicRequest
-            .header("Accept", "application/json")
-            .post(credentials.tokenUri)
-            .body(body)
-            .response(asJson[ClientCredentialsResponse])
-            .send(sttpBackend)
-          payload <- response.body match {
-            case Right(response) => F.pure(response)
-            case Left(responseException) => // Non-2xx response
-              responseException match {
-                case HttpError(body, statusCode) =>
-                  F.raiseError[ClientCredentialsResponse](
-                    new SdkException(
-                      body,
-                      uri = Some(credentials.tokenUri),
-                      responseCode = Some(statusCode.code)
-                    )
-                  )
-                case DeserializationException(_, error) =>
-                  F.raiseError(
-                    new SdkException(
-                      s"Failed to parse response from IdP: ${error.getMessage}"
-                    )
-                  )
-              }
+          now <- clock.monotonic
+          res <- maybeCacheToken match {
```

Comment:
"res" is sort of ok as a variable name with http4s responses, but it's not helpful here. `newToken`?

Reply:
changed to `newToken`